src/Nancy.Authentication.Forms.Tests/FormsAuthenticationFixture.cs

```diff
@@ -162,7 +161,71 @@ public void Should_return_ok_response_when_user_logs_in_without_redirect()
             result.StatusCode.ShouldEqual(HttpStatusCode.OK);
         }
 
-        [Fact]
+		#region Throw helpful exception when the configuration is not enabled
+
+		[Fact]
+		public void Should_throw_helpful_exception_message_when_user_logs_in_without_redirect_and_forms_authentication_not_enabled()
+		{
+			// Given
+			const string expectedMessage = "The internal FormsAuthenticationConfiguration has not been set. Ensure that FormsAuthentication has been enabled in the bootstrapper";
+			FormsAuthentication.Disable();
+
+			// When
+			var result = Record.Exception(() => FormsAuthentication.UserLoggedInResponse(userGuid));
+
+			// Then
+			result.ShouldBeOfType(typeof(InvalidOperationException));
+			result.Message.ShouldBeSameAs(expectedMessage);
+		}
+
+		[Fact]
+		public void Should_throw_helpful_exception_message_when_user_logs_in_with_redirect_and_forms_authentication_not_enabled()
+		{
+			// Given
+			const string expectedMessage = "The internal FormsAuthenticationConfiguration has not been set. Ensure that FormsAuthentication has been enabled in the bootstrapper";
+			FormsAuthentication.Disable();
+
+			// When
+			var result = Record.Exception(() => FormsAuthentication.UserLoggedInRedirectResponse(context, userGuid));
+
+			// Then
+			result.ShouldBeOfType(typeof(InvalidOperationException));
+			result.Message.ShouldBeSameAs(expectedMessage);
+		}
+
+		[Fact]
+		public void Should_throw_helpful_exception_message_when_user_logs_out_with_redirect_and_forms_authentication_not_enabled()
+		{
+			// Given
+			const string expectedMessage = "The internal FormsAuthenticationConfiguration has not been set. Ensure that FormsAuthentication has been enabled in the bootstrapper";
+			FormsAuthentication.Disable();
+
+			// When
+			var result = Record.Exception(() => FormsAuthentication.LogOutAndRedirectResponse(context, "/"));
+
+			// Then
+			result.ShouldBeOfType(typeof(InvalidOperationException));
+			result.Message.ShouldBeSameAs(expectedMessage);
+		}
+	
+		[Fact]
+		public void Should_throw_helpful_exception_message_when_user_logs_out_without_redirect_and_forms_authentication_not_enabled()
+		{
+			// Given
+			const string expectedMessage = "The internal FormsAuthenticationConfiguration has not been set. Ensure that FormsAuthentication has been enabled in the bootstrapper";
+			FormsAuthentication.Disable();
+
+			// When
+			var result = Record.Exception(() => FormsAuthentication.LogOutResponse());
+
+			// Then
+			result.ShouldBeOfType(typeof(InvalidOperationException));

```

Comment:
Please change this to use the generic overload `result.ShouldBeOfType<InvalidOperationException>();`